src/qt/evtloop.cpp

```diff
@@ -59,35 +74,45 @@ void wxQtIdleTimer::idle()
         wxTheApp->ProcessPendingEvents();
 
     // Send idle event
-    if ( m_eventLoop->ProcessIdle() )
-        m_eventLoop->ScheduleIdleCheck();
+    if (wxTheApp->ProcessIdle())
+        ScheduleIdleCheck();
+}
+
+
+namespace
+{
+    wxQtIdleTimer *gs_idleTimer = NULL;
+}
+
+void wxQtIdleTimer::ScheduleIdleCheck()
+{
+    wxQtEventLoopBase *eventLoop = static_cast<wxQtEventLoopBase*>(wxEventLoop::GetActive());
+    if ( eventLoop )
+        eventLoop->ScheduleIdleCheck();
 }
 
 wxQtEventLoopBase::wxQtEventLoopBase()
 {
-    // We need a QCoreApplication for event loops, create it here if it doesn't
-    // already exist as we can't modify wxAppConsole
-    if ( !QCoreApplication::instance() )
+    // Create an idle timer to run each time there are no events (timeout = 0)
+    if ( !gs_idleTimer )
     {
-        new QApplication( wxAppConsole::GetInstance()->argc, wxAppConsole::GetInstance()->argv );
+        gs_idleTimer = new wxQtIdleTimer();
+    }
+    else
+    {
+        gs_idleTimer->IncRef();
     }
-    
-    // Create an idle timer to run each time there are no events (timeout = 0)
-    m_qtIdleTimer = new wxQtIdleTimer( this );
-
-    // Pass all events to the idle timer, so it can be restarted each time
-    // an event is received
-    qApp->installEventFilter( m_qtIdleTimer );
-
 
+    m_qtIdleTimer = gs_idleTimer;
     m_qtEventLoop = new QEventLoop;
 }
 
 wxQtEventLoopBase::~wxQtEventLoopBase()
 {
-    qApp->removeEventFilter(m_qtIdleTimer);
+    if ( gs_idleTimer->GetRefCount() <= 1 )
```

Comment:
We should almost certainly call `DecRef()` here, shouldn't we?

Reply:
The idea is that there should only be one idle timer regardless of how many event loops exist but we don’t want a idle timer to exist if there are no event loops. Each event loop instance stores as reference to the timer wxObjectDataPtr ensures the destruction of an event loop automatically decreases the ref count.  The last one to be destroyed NULL’s the global instance.

Reply:
OK, sorry, I was confused by the combined use of `wxObjectDataPtr` and explicit `IncRef()` calls. I now see that it indeed looks correct, but I think it would be still better to avoid mixing them up: could we please use `wxObjectDataPtr` for `gs_idleTimer` too? This should avoid the need for any explicit `IncRef()`s.